databricks/sdk/core.py

```diff
@@ -91,12 +91,24 @@ def inner() -> Dict[str, str]:
 @credentials_provider('runtime', [])
 def runtime_native_auth(cfg: 'Config') -> Optional[HeaderFactory]:
     from databricks.sdk.runtime import init_runtime_native_auth
-    if init_runtime_native_auth is None:
-        return None
-    else:
+    if init_runtime_native_auth is not None:
         host, inner = init_runtime_native_auth()
         cfg.host = host
         return inner
+    try:
+        from dbruntime.databricks_repl_context import get_context
+        ctx = get_context()
+        if ctx is None:
+            logger.debug('Empty REPL context returned, skipping runtime auth')
+            return None
+        cfg.host = f'https://{ctx.browserHostName}'
+
+        def inner() -> Dict[str, str]:
+            return {'Authorization': f'Bearer {ctx.apiToken}'}
```

Comment:
Don't we lock in the `apiToken`? I think we should call `get_context()` inside `inner` to swap the context for different users.

Reply:
@xiaochen-db also running `get_context()` within the `inner`? thanks for pointing this out! will add

Reply:
addressed in https://github.com/databricks/databricks-sdk-py/pull/152/commits/b95af594b9d0ee519e03ebc584c1aeac1f1f51f3